src/protocolsupport/protocol/typeremapper/itemstack/complex/toclient/EnchantToPEEnchantSpecificRemapper.java

```diff
@@ -18,19 +16,15 @@
 	public NetworkItemStack remap(ProtocolVersion version, String locale, NetworkItemStack itemstack) {
 		if ((itemstack.getNBT() != null)) {
 			NBTCompound tag = itemstack.getNBT();
-			NBTList<NBTCompound> ench = tag.getTagListOfType("ench", NBTType.COMPOUND);
+			NBTList<NBTCompound> ench = tag.getTagListOfType(CommonNBT.MODERN_ENCHANTMENTS, NBTType.COMPOUND);
 			if (ench != null) {
-				tag.setTag("ench", remapEnchantList(ench));
+				tag.setTag(CommonNBT.LEGACY_ENCHANTMENTS, remapEnchantList(ench));
 			}
-			NBTList<NBTCompound> stored_enchants = tag.getTagListOfType("stored-enchants", NBTType.COMPOUND);
+			NBTList<NBTCompound> stored_enchants = tag.getTagListOfType(CommonNBT.BOOK_ENCHANTMENTS, NBTType.COMPOUND);
 			if (stored_enchants != null) {
-				tag.setTag("stored-enchants", remapEnchantList(stored_enchants));
+				tag.setTag(CommonNBT.LEGACY_ENCHANTMENTS, remapEnchantList(stored_enchants));
```

Comment:
This doesn't seem right...? Can you check if you got the right names here?

Reply:
so... this is how i ended up getting enchanted books to work :\ I just use the normal legacy enchant NBT field and it works fine. I couldnt get any of the other stored-enchantment like fields to work. They look exactly like that do in single player, and work fine. Even in single player all the enchanted books say 'Items' instead of the specific type they work for. Eh, once we get the creative inventory NBT working this should also be easy to track down